### PR TITLE
Added tests to message pagination to reach 100% code coverage.

### PR DESCRIPTION
**Pull request by illiriks**
Tests were added by creating another set of tests and a `nock` that sent a link for the next page. Used exactly the same messages as before. Will proceed to refactor those tests somewhat after it passes build on Travis.
Moved part of `lib/client.js` in order to resolve build failure caused by `jslint`.

### CODE REVIEW

**Comment by illiriks** [comment]
Previous tests worked on some Node versions, but not all. 
In order to use deep-equals, I had to switch to the `expect()` syntax. The old tests were somewhat verbose, but more importantly, they didn't test all of the fields of the message. I can switch it back if stylistic consistency is important. 

Any and all feedback would be appreciated.

File: package.json [hunk 1]
```diff
@@ -26,11 +26,12 @@
     "request": "^2.69.0"
   },
   "devDependencies": {
+    "chai": "^3.5.0",
```
Comment (dtolb) [inline package.json]: Old test were done with [should](https://github.com/bandwidthcom/node-bandwidth/pull/53/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R43) If we're going to move to chai & `expect` we **_should_** talk about that to make sure we're down with updating all the tests.

Also @nbragdon had mentioned reevaluating test framework

Comment (illiriks) [inline package.json]: Good point. I confused the use of `should.js` with the 'should' style of Chai assertions. Fixed in commit 3d145f3.

**Comment by illiriks** [comment]
Back to `should.js`. I got confused about test frameworks :'(

File: lib/client.js [hunk 2]
```diff
@@ -12,6 +12,21 @@ var Client = function (config) {
 		config.baseUrl = "https://api.catapult.inetwork.com";
 	}
 
+	function handleResponse(response) {
```
Comment (dtolb) [inline lib/client.js:15]: any reason for change?

Comment (illiriks) [inline lib/client.js:15]: The linter was complaining about using the function before it was defined, and it was breaking build. Moving the declarations of `handleResponse` and `getUserAgentHandler` fixed build.

Comment (dtolb) [inline lib/client.js:15]: ok, that's what I thought.

**Pull request merged**